command_line/dials_import.py

```diff
@@ -308,15 +308,14 @@ def __call__(self, imageset):
     Override the parameters
 
     '''
-    from dxtbx.imageset import ImageSet
     from dxtbx.imageset import ImageSweep
     from dxtbx.model import BeamFactory
     from dxtbx.model import DetectorFactory
     from dxtbx.model import GoniometerFactory
     from dxtbx.model import ScanFactory
     from copy import deepcopy
     if self.params.geometry.convert_sweeps_to_stills:
-      imageset = ImageSet(data=imageset.data())
+      imageset = imageset.__class__(data=imageset.data())
```

Comment:
should this be passed say an *imagesetfactory* function rather than a class instance?
(If I was to subclass imageset and change the ```__init__``` function this will blow up, right?)

Reply:
We have subclassed imageset (ImageSetJIT), which is why this change is needed.  There is no factory function of ImageSet that takes ImageSetData instead of filenames, and if there were, it would have to pick between ImageSet, ImageSetJIT and MemImageSet according to some criteria.  Since we already know what the correct class is when ManualGeometryUpdater is called, we can just create a new instance of the class instead.